extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -221,48 +223,66 @@ class QueueProcessor {
 
     _getData(entry, cb) {
         this.log.debug('getting data', { entry: entry.getLogInfo() });
-        const req = this.S3source.getObject({
-            Bucket: entry.getBucket(),
-            Key: entry.getObjectKey(),
-            VersionId: entry.getEncodedVersionId() });
-        const incomingMsg = req.createReadStream();
-        req.on('error', err => {
-            this.log.error('error response getting data from S3',
-                           { entry: entry.getLogInfo(),
-                             origin: this.sourceConfig.s3,
-                             error: err.message,
-                             httpStatus: err.statusCode });
-            incomingMsg.emit('error', err);
+        const locations = entry.getLocation();
+        return async.timesLimit(locations.length, LIMIT, (i, next) => {
```

Comment:
@bennettbuchanan apparently you don't need `i` to get the part number since you're calling `entry.getPartNumber()` ;)

To get the array of results at the end, it seems like `mapLimit` would be the right answer here. But if we go towards doing the perf optimization I mentioned, keeping the whole list of streams will not be needed anymore.